File: C++/0036. Calculator Ver.2.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    string s;
    while(cin>>s){
        int res = 0;
        int flag = 1;
        for(auto c:s){
            if(c=='('){
                ++res;
            }
            else if(c==')'){
                if(res==0){
                    flag = 0;
                    break;
                }
                --res;
            }
        }
        cout<<((res!=0||flag==0)?"NO":"YES")<<endl;
    }
    return 0;
}

File: Python/0036. Calculator Ver.2.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            s = input().strip()
            res = 0
            flag = 1
            for c in s:
                if c == '(':
                    res += 1
                elif c == ')':
                    if res == 0:
                        flag = 0
                        break
                    res -= 1
            print("NO" if res!=0 or flag == 0 else "YES")
            
        except EOFError:
            break
    
if __name__ == "__main__":
    func()
